linker/shared/src/main/scala/org/scalajs/linker/backend/javascript/Printers.scala

```diff
@@ -34,8 +35,39 @@ import Trees._
  * hotspots in this object.
  */
 object Printers {
+  def printFile(tree: Tree): ByteBuffer = {
+    val byteStream = new ByteArrayOutputStream
+    val out = new OutputStreamWriter(byteStream, StandardCharsets.UTF_8)
+
+    val printer = new JSTreePrinter(out)
+    printer.printTopLevelTree(tree)
+
+    out.close()
+    ByteBuffer.wrap(byteStream.toByteArray())
+  }
+
+  def printFileWithSourceMap(tree: Tree,
+      sourceMapURI: Option[URI],
+      jsFileURI: Option[URI],
+      relativizeBaseURI: Option[URI]): (ByteBuffer, ByteBuffer) = {
+
+    val byteStream = new ByteArrayOutputStream
+    val out = new OutputStreamWriter(byteStream, StandardCharsets.UTF_8)
+
+    val sourceMap = new SourceMapWriter(jsFileURI, relativizeBaseURI)
+
+    val printer = new JSTreePrinterWithSourceMap(out, sourceMap)
+    printer.printTopLevelTree(tree)
+    sourceMapURI.foreach(uri =>
+        out.write("//# sourceMappingURL=" + uri.toASCIIString() + "\n"))
 
-  class JSTreePrinter(protected val out: Writer) extends IndentationManager {
+    out.close()
+
+    (ByteBuffer.wrap(byteStream.toByteArray()), sourceMap.result())
+  }
+
+  private[javascript] class JSTreePrinter(protected val out: Writer)
```

Comment:
scalajs-bundler uses this class, although it really shouldn't. Can we keep it public?